src/command/restore/restore.c

```diff
@@ -1248,9 +1300,20 @@ restoreSelectiveExpression(Manifest *manifest)
             {
                 const String *db = strLstGet(dbList, dbIdx);
 
-                // Only user created databases can be zeroed, never system databases
-                if (cvtZToUInt64(strZ(db)) >= PG_USER_OBJECT_MIN_ID)
+                // Include system databases by default
+                if (strLstExists(excludeDbIdList, db) ||
+                	(strLstSize(includeDbIdList) > 0 && cvtZToUInt64(strZ(db)) >= PG_USER_OBJECT_MIN_ID && 
+                	!strLstExists(includeDbIdList, db) && !strLstExists(systemDbIdList, db)))
```

Comment:
I think we have more String Lists than we need. The end result is that we want dbList to contain the IDs of the databases that we will zero. The problem with the original code is that it kept the system databases in the dbList and then thought they could just be filtered out by checking their ID. That, of course, is the bug. So our end goal should be that the dbList must only contain the IDs of the databases to zero. So first we should check  that the user does not provide the same database in the --db-include and --db-exclude list (as you did - good job!). Then any --db-exclude databases must **remain** in the dbList. Any system database that is **not** in the exclude list, must be **removed** from the dbList. Any --db-include database must be **removed** from the list. Then the entire IF condition goes away here because all the databases in the dbList are going to be zeroed.

Reply:
Hm, is it a memory/perf issue to have an extra small string list to make the code more readable ?

Since we have to keep track of the databases to exclude (via dblist minus include-list or via exclude-list), I didn't managed to remove more than the `includeDbIdList`. 
See commit f494c9a about it.

I hope this is approaching what you had in mind :-)